samples/apps/copilot-chat-app/webapi/CopilotChat/Controllers/DocumentImportController.cs

```diff
@@ -110,9 +114,26 @@ private enum SupportedFileType
                     return this.BadRequest($"Unsupported file type: {fileType}");
             }
 
-            await this.ParseDocumentContentToMemoryAsync(kernel, fileContent, documentImportForm);
+            var memorySource = new MemorySource(
+                documentImportForm.ChatId.ToString(),
+                formFile.FileName,
+                documentImportForm.UserId,
+                MemorySourceType.File,
+                null);
+
+            await this._sourceRepository.UpsertAsync(memorySource);
+
+            try
+            {
+                await this.ParseDocumentContentToMemoryAsync(kernel, fileContent, documentImportForm, memorySource.Id);
+            }
+            catch (Exception exception)
+            {
+                await this._sourceRepository.DeleteAsync(memorySource);
+                throw exception;
```

Comment:
This will reset the callstack - I recommend just calling throw.
```suggestion
                throw;
```